File: src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionMatchAllTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import org.junit.jupiter.api.Test;

import java.util.Arrays;

/**
 * @see TransformerFunctionMatchAll
 */
public class TransformerFunctionMatchAllTest extends BaseTest {
    @Test
    void testInlineFunctionMatch() {
        var input = "hello my helloKitty";
        assertTransformation(input, "$$matchall([el]):$", Arrays.asList("e", "l", "l", "e", "l", "l"));
        assertTransformation(input, "$$matchall([le]+):$", Arrays.asList("ell", "ell"));
        assertTransformation(input, "$$matchall(hell):$", Arrays.asList("hell","hell"));
        assertTransformation(input, "$$matchall(^hello):$", Arrays.asList("hello"));
        assertTransformation(input, "$$matchall(hello$):$", null);
    }
    @Test
    void inlineGroup() {
        assertTransformation("world to waterWorld", "$$matchall('w(\\\\w+)d',1):$", Arrays.asList("orl", "aterWorl"));
    }
}

File: javascript/json-transform/src/__tests__/functions/TransformerFunctionMatchAll.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";

describe("TransformerFunctionMatchAll", () => {
  test("inline", async () => {
    const input = "hello my helloKitty";
    await assertTransformation(input, "$$matchall([el]):$", ["e", "l", "l", "e", "l", "l"]);
    await assertTransformation(input, "$$matchall([le]+):$", ["ell", "ell"]);
    await assertTransformation(input, "$$matchall(hell):$", ["hell", "hell"]);
    await assertTransformation(input, "$$matchall(^hello):$", ["hello"]);
    await assertTransformation(input, "$$matchall(hello$):$", null);
  });

  test("inlineGroup", async () => {
    await assertTransformation("world to waterWorld", "$$matchall('w(\\\\w+)d',1):$", ["orl", "aterWorl"]);
  });
});
